internal/controllers/addon/olm_reconciler.go

```diff
@@ -21,6 +21,7 @@ type olmReconciler struct {
 	client                  client.Client
 	uncachedClient          client.Client
 	operatorResourceHandler operatorResourceHandler
+	observedSubscription    operatorsv1alpha1.Subscription
```

Comment:
To answer your questions @pepedocs 
1. No, `olmReconciler` is not concurrency-safe
2. There is not a 1:1 relationship between `olmReconciler` instances and addons, but rather one instance of `olmReconciler` is globally used for all addons. So what you really would want here is a concurrency-safe map from addons to subscriptions.
3. While an additional map would work the clients, caches, and informers supporting them will already prevent unnecessary additional API calls by returning cached objects and keeping them up-to-date. So an additional cache just for the `olmReconciler` would be redundant. A simple `get` call in `phaseObserveOperatorResource` would do.